AndroidSDKCore/src/main/java/com/leanplum/internal/Request.java

```diff
@@ -23,51 +23,81 @@
 
 import com.leanplum.Leanplum;
 
+import org.json.JSONObject;
+
+import java.util.HashMap;
 import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Leanplum request class.
+ *
+ * @author Andrew First
+ */
+public class Request {
 
-public class Request implements Requesting {
+  private String requestId = UUID.randomUUID().toString();
 
   private final String httpMethod;
-  private final String apiMethod;
+  private final String apiAction;
   private final Map<String, Object> params;
-  private ResponseCallback response = null;
-  private ErrorCallback error = null;
-  private final CountAggregator countAggregator;
+  ResponseCallback response;
+  ErrorCallback error;
+  private boolean sent;
+
+  public String requestId() {
+    return requestId;
+  }
 
-  public Request(String httpMethod, String apiMethod, Map<String, Object> params) {
+  public Request(String httpMethod, String apiAction, Map<String, Object> params) {
     this.httpMethod = httpMethod;
-    this.apiMethod = apiMethod;
-    this.params = params;
-    this.countAggregator = Leanplum.countAggregator();
+    this.apiAction = apiAction;
+    this.params = params != null ? params : new HashMap<>();
+    // Check if it is error and here was SQLite exception.
+    if (RequestBuilder.ACTION_LOG.equals(apiAction) && LeanplumEventDataManager.sharedInstance().willSendErrorLogs()) {
```

Comment:
This should be moved to request sender instead of checking it here. 
We can do it another PR

Reply:
It is better to do this in another PR.